.github/workflows/actions.yml

```diff
@@ -19,11 +15,26 @@ jobs:
         os: [ubuntu-latest, windows-latest, macos-latest]
     name: ${{ matrix.os }} - Java ${{ matrix.java }}
     steps:
-      - uses: actions/checkout@v2
+      - name: Check out repo
+        uses: actions/checkout@v2
+        # fetch unshallow to enable blame for Sonar
+      - name: Fetch unshallow
```

Comment:
not sure if we need an extra step we could also inline this into the checkout step - https://github.com/actions/checkout#fetch-all-history-for-all-tags-and-branches

Reply:
AFAIK the first step does the actual checkout with a bunch of optimizations:

https://github.com/beatngu13/junit-pioneer/runs/551399103?check_suite_focus=true#step:2:1

Whereas the second step only fetches the history:

https://github.com/beatngu13/junit-pioneer/runs/551399103?check_suite_focus=true#step:3:1

I don't think we can use one without the other? Unless we do the checkout manually and avoid setting depth to 1 (like [here](https://github.com/beatngu13/junit-pioneer/runs/551399103?check_suite_focus=true#step:2:26)).

Probably related to this feature request:

https://github.com/actions/checkout/issues/204

Reply:
maybe i miss understood the logic in the readme of the action
```
- uses: actions/checkout@v2
- run: |
    git fetch shallow--prune --un
```

i thought based on this we can just do

```
  - name: Check out repo
        uses: actions/checkout@v2
        run: |
             git fetch shallow--prune --un
```

But it also seams like i miss read yaml once again, and ignored the `-` which i just realized by writing this comment :)